src/v/kafka/server/handlers/details/security.h

```diff
@@ -307,4 +312,104 @@ inline ss::sstring to_kafka_host(security::acl_host host) {
     }
 }
 
+/**
+ * In some Kafka APIs, set of allowed operations is encoded as a int32 bit
+ * field. i.e. each allowed operation corresponds to `1` at position defined by
+ * operation underlying value.
+ *
+ * f.e.
+ * allowed read, write and alter would be
+ *
+ *  00000000 00000000 00000000 10001100
+ */
+inline int32_t to_bit_field(const std::vector<security::acl_operation>& ops) {
+    static constexpr int available_bits = std::numeric_limits<int32_t>::digits
+                                          - 1;
```

Comment:
You don't need the `-1`, `digits` does not include the sign bit for signed types.

Reply:
perfect, will change it